### PR TITLE
Fix for `ignores` not working when base doesn't equal cwd

### LINKED ISSUE #9: Ignores fail when used in a subdirectory
First of all, thanks for all your hard work on this! I really appreciate it. 
I'm having a problem where gulp-xo seems to fail when dealing with ignores that have a subfolder. 
Example: 

``` json
//Doesn't work.
"ignores": [
  "src/vendor/**"
]
//Works.
"ignores": [
  "vendor/**"
]
```

[The regular xo tool works fine, ignoring the files and continuing on. ](https://cloud.githubusercontent.com/assets/6267043/18460647/d4ace55e-7938-11e6-8147-a376557d77b1.png)
[On the other hand, the gulp-xo tool will still check these files. ](https://cloud.githubusercontent.com/assets/6267043/18460648/d4acde56-7938-11e6-91c3-540102ace300.png)

Those pictures were both taken, one after the other, in the same development enviroment.

Gulpfile:

``` js
const gulp = require('gulp');
const xo = require('gulp-xo');

gulp.task('xo', function () {
    return gulp.src('src/**/*.js')
    .pipe(xo())
    .pipe(gulp.dest('.tmp'));
});
```

package.json:

``` json
"xo": {
    "envs": [
      "browser"
    ],
    "globals": [
      "angular",
      "_"
    ],
    "ignores": [
      "gulp_tasks/**",
      "conf/**",
      "src/vendor/**"
    ]
  }
```

I'm sorry if I missed something obvious about gulp-xo and the difference from xo.

Thanks for your work on this project and other projects.

### PR DESCRIPTION
**Pull request by gucong3000**
fix #9

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -13,7 +13,7 @@
     "node": ">=4"
   },
   "scripts": {
-    "test": "xo --ignore='_fixture.js' && ava"
+    "test": "xo --ignore _fixture.js && ava"
```
Comment (gucong3000) [inline package.json]: Compatible Windows

File: index.js [hunk 2]
```diff
@@ -28,7 +29,7 @@ module.exports = opts => {
 		try {
 			report = xo.lintText(file.contents.toString(), {
 				cwd: file.cwd,
-				filename: file.relative
+				filename: path.relative(file.cwd, file.path)
```
Comment (sindresorhus) [inline index.js]: `file.basename` would be the correct thing to use here

File: package.json (hunk 1 above)
Comment (sindresorhus) [inline package.json]: You don't have to remove the `=`, just the single-quotes.

File: index.js (hunk 2 above)
Comment (gucong3000) [inline index.js]: No, I'm pretty sure. here should be `path.relative(file.cwd, file.path)`

File: package.json (hunk 1 above)
Comment (gucong3000) [inline package.json]: I am working on Windows, single-quotes is not work on Windows 10.

File: index.js (hunk 2 above)
Comment (sindresorhus) [inline index.js]: Why is that your opinion? `file.basename` is the filename part of the path, which is exactly what `xo.lintText` expects.

https://github.com/gulpjs/vinyl#filebasename

File: package.json (hunk 1 above)
Comment (sindresorhus) [inline package.json]: Yes, but the `=` does work.

File: index.js (hunk 2 above)
Comment (gucong3000) [inline index.js]: https://github.com/gulpjs/vinyl/blob/master/index.js#L256

Comment (gucong3000) [inline index.js]: We are all wrong, we should use absolute path.

Comment (sindresorhus) [inline index.js]: > gulpjs/vinyl:index.js@master#L256

Not sure what you're trying to say by linking this, but I do realize how it works underneath.

> We are all wrong, we should use absolute path.

No, but an absolute path works too.

**Pull request merged**